### PR TITLE
Add array support to run

### PR DESCRIPTION
**Pull request by kayhhh**
untested

### CODE REVIEW

File: src/schedule/run.ts [hunk 1]
```diff
@@ -17,8 +17,12 @@ class SystemConfig {
 	 * @param ...systems The systems that this system must run before.
 	 * @returns `this`, for chaining.
 	 */
-	before(...systems: System[]): this {
-		this.dependents.push(...systems);
+	before(...systems: (System | System[])[]): this {
+		for (const system of systems) {
+			if (Array.isArray(system)) this.dependents.push(...system);
```
Comment (JaimeGensler) [inline src/schedule/run.ts:22]: Stylistic nit: Would prefer not using single-line if/else here (and elsewhere), braces/blocks help me parse control flow more quickly and is more consistent with the rest of the codebase 🙂

File: src/schedule/run.ts [hunk 2]
```diff
@@ -62,14 +70,24 @@ class SystemConfig {
...
-const run: Run = system => new SystemConfig(system);
+const run: Run = system =>
+	Array.isArray(system)
+		? system.map(s => new SystemConfig(s))
+		: new SystemConfig(system);
+
 run.chain = (...systems) =>
 	systems.map((system, i) =>
-		i === 0 ? system : new SystemConfig(system).after(systems[i - 1]),
+		i === 0
```
Comment (JaimeGensler) [inline src/schedule/run.ts:86]: Style nit: would prefer to remove this nested ternary. Could just move this first case (`i === 0`) to an `if` and then use a (no longer nested) ternary in the return

File: src/schedule/run.ts [hunk 3]
```diff
@@ -62,14 +70,24 @@ class SystemConfig {
 export { SystemConfig };
 
 type Run = {
-	(system: System): SystemConfig;
-	chain(...systems: System[]): (System | SystemConfig)[];
+	(system: System | System[]): SystemConfig | SystemConfig[];
+	chain(
+		...systems: (System | System[])[]
+	): (System | System[] | SystemConfig | SystemConfig[])[];
```
Comment (JaimeGensler) [inline src/schedule/run.ts:76]: I think we'll end up using this `(System | System[] | SystemConfig | SystemConfig[])[]` type a lot - we could extract it to its own type & export here (e.g. `SystemList`, or whatever sounds best), for use elsewhere, too

**Review by JaimeGensler — CHANGES_REQUESTED** [review]
Looks great overall - thanks so much for taking care of this! Left a couple style nits to keep things inline with the rest of the codebase.
Lack of test setup here is my bad once again, I'll get to it - if you could leave a simple "TODO: Tests" comment at the bottom of this file that would be super helpful for me 😄 

I know this expands the scope here a bit, but if you're willing (& agree with my thinking), it would probably be good if `addSystems()` could accept System/SystemConfig arrays without spreading, too. If you'd rather be done here, I'll handle that myself 🙂 

Thanks again!

**Comment by kayhhh** [comment]
I can add `addSystems()` array support, probably good to have all system APIs like this to support the same syntax of accepting both plain systems and arrays.

**Comment by kayhhh** [comment]
Will assume this includes `addSystemsToSchedule()`

**Comment by kayhhh** [comment]
Hm, so I did find an issue with the `run` changes I added. Using arrays like `run([systemA]).last()` throws an error, because run is returning `SystemConfigs[]`, not a singular config. I can just remove array support here unless you can think of a way to resolve it. `run.chain` should still work fine.

Also sidenote, is there a reason for `dist` to be committed? I typically run build to catch errors like this (and it is how I found it now), but its a bit annoying to have to remove my files after the build so they aren't committed.

**Comment by JaimeGensler** [comment]
>  I can just remove array support here unless you can think of a way to resolve it

Wondering if it's reasonable for `SystemConfig` itself to accept multiple systems, so run could pass the array directly to `SystemConfig` and all systems can be configured together. There may be some issues with this I haven't thought of, but it's a potential solution.

> Also sidenote, is there a reason for dist to be committed?

Originally I thought it'd make it easier to keep an eye on build output for anything problematic/odd, but I've also felt lately it's more tedious than helpful. Feel free to add it to the .gitignore here & delete the folder in this PR, or I'll do it in a follow-up.

**Comment by JaimeGensler** [comment]
That makes sense! Expanding support also brings up a lot of questions around things like permitted depth of nesting, which requires a bit more thought. Going to re-review a little later and should be able to get this merged, thanks! 🙌

**Comment by kayhhh** [comment]
Sounds good. I removed array support from `run(...)` since it was causing issues, should be working for `run.chain`, `.after`, and `.before`.

**Review by JaimeGensler — APPROVED** [review]
Looks great! Thanks for the revert on `run()`, I think it's something that'll be good to do eventually but there's definitely some questions that need answering first.

**Pull request merged**